crates/cli-support/src/js/mod.rs

```diff
@@ -3305,6 +3314,47 @@ impl<'a> Context<'a> {
             format!("{}{}", name, cnt)
         }
     }
+
+    fn require_stack_pointer_shim(&mut self) {
+        self.should_generate_stack_pointer_shim = true;
+    }
+
+    fn inject_stack_pointer_shim(&mut self) -> Result<(), Error> {
+        use walrus::ir::*;
+
+        let mut builder =
+            walrus::FunctionBuilder::new(&mut self.module.types, &[ValType::I32], &[ValType::I32]);
+        builder.name("__wbindgen_add_to_stack_pointer".to_string());
+
+        let mut body = builder.func_body();
+        let arg = self.module.locals.add(ValType::I32);
+        let local = self.module.locals.add(ValType::I32);
+
+        let stack_pointer = match self.aux.shadow_stack_pointer {
+            Some(s) => s,
+            // In theory this shouldn't happen since malloc is included in
+            // most wasm binaries (and may be gc'd out) and that almost
+            // always pulls in a stack pointer. We can try to synthesize
+            // something here later if necessary.
+            None => bail!("failed to find shadow stack pointer"),
+        };
+
+        // Create a shim function that mutate the stack pointer
+        // to avoid exporting a mutable global.
+        body.local_get(arg)
+            .global_get(stack_pointer)
+            .binop(BinaryOp::I32Add)
+            .local_tee(local)
+            .global_set(stack_pointer)
+            .local_get(local);
+
+        let add_to_stack_pointer_func = builder.finish(Vec::new(), &mut self.module.funcs);
```

Comment:
I think the `Vec::new()` here is the list of arguments, right? And `arg` should be the one-element vector here?